Lib/ipaddress.py

```diff
@@ -1834,8 +1855,25 @@ def __init__(self, address):
         addr_str = str(address)
         if '/' in addr_str:
             raise AddressValueError("Unexpected '/' in %r" % address)
+        addr_str, self._scope_id = self._split_scope_id(addr_str)
+
         self._ip = self._ip_int_from_string(addr_str)
 
+    def __str__(self):
+        ip_str = super().__str__()
+        return ip_str + '%' + self._scope_id if self._scope_id else ip_str
```

Comment:
Could you clarify you suggestion, please?

Reply:
Sure -- the request is this bit:

> To help give notice to third-party code, we should be sure to make the scoped-address case visible in the docs and in What's New.

Concretely that means something like:
* This new feature should be mentioned in the "What's New in 3.9" document. That's `Doc/whatsnew/3.9.rst`. (After all, it's one of the nice new things we'll have in 3.9!)
* Each of the main places in the docs that have a series of examples should include an example with a scoped address, so that it's visible that these addresses with `%` in them are a thing that can happen. That means something like:
  * The initial "Convenience factory functions" section in the module doc. For a lot of people this may be the one section they really read. One example in either the `ip_address` or the `ip_network` case should be enough.
  * The docs bit on the `IPv6Address` constructor. Already there in this version -- thanks!
  * The "Conversion to Strings and Integers" section in the doc.
  * The first block of examples in the "IP Host Addresses" section at the top of `Doc/howto/ipaddress.rst` aka [here](https://docs.python.org/3/howto/ipaddress.html). That's prominently linked to from the module doc, so a lot of people may rely more on that.

Reply:
@gnprice , great thanks, really very helpful. In progress.